### PR TITLE
AK+Piano+LibGUI: Prepare for per-track controls

### PR DESCRIPTION
**Pull request by kleinesfilmroellchen**
This is an assortment of bug fixes, layouting improvements, and small Piano refactors in order to prepare for per-track controls in Piano.

The LibGUI fixes make dynamic layouting work better with labels and sliders, as you can see in these screenshots:

### CODE REVIEW

**Comment by frhun** [comment]
This has some overlap with [set_fixed_size](https://github.com/SerenityOS/serenity/pull/17407).

File: Userland/Libraries/LibGUI/Slider.cpp [hunk 1]
```diff
@@ -33,6 +33,14 @@ Optional<UISize> Slider::calculated_min_size() const
     size.set_primary_size_for_orientation(orientation(), track_margin() * 2 + knob_fixed_primary_size());
     return size;
 }
+Optional<UISize> Slider::calculated_preferred_size() const
+{
+
+    UISize size { 0, 0 };
+    size.set_secondary_size_for_orientation(orientation(), calculated_min_size()->secondary_size_for_orientation(orientation()));
```
Comment (frhun) [inline Userland/Libraries/LibGUI/Slider.cpp:40]: Just setting it to `SpecialDimension::Shrink` is the more ideomatic way of achieving this.

**Comment by kleinesfilmroellchen** [comment]
@frhun I think I removed all conflicting layout changes and it looks fine on master now that your PR is merged.

File: Userland/Applications/Piano/MainWidget.cpp [hunk 2]
```diff
@@ -58,6 +59,28 @@ ErrorOr<void> MainWidget::initialize()
 
     m_keys_widget = TRY(m_keys_and_knobs_container->try_add<KeysWidget>(m_track_manager.keyboard()));
 
+    m_octave_container = TRY(m_keys_and_knobs_container->try_add<GUI::Widget>());
+    m_octave_container->set_preferred_width(GUI::SpecialDimension::Fit);
+    (void)TRY(m_octave_container->try_set_layout<GUI::VerticalBoxLayout>());
```
Comment (AtkinsSJ) [inline Userland/Applications/Piano/MainWidget.cpp:64]: The (void) here is no longer needed.

**Review by AtkinsSJ — CHANGES_REQUESTED** [review]
From a fairly quick phone review, this looks good, I just have one nitpick.

**Pull request merged**